proof/invariant-abstract/AARCH64/ArchInvariants_AI.thy

```diff
@@ -686,10 +686,19 @@ locale_abbrev valid_global_tables :: "'z::state_ext state \<Rightarrow> bool" wh
 
 lemmas valid_global_tables_def = valid_global_tables_2_def
 
+(* The number of list registers should not exceed the maximum word value *)
```

Comment:
This doesn't really help, and the commit it comes in says "add numlistregs invariant" which also doesn't match the name of this invariant... I'm not really sure why it's fine to have a bound of `2^word_bits` anyhow.

Reply:
Why not? (they are `nat`)

Reply:
Getting and setting list registers involves `word` to `nat` conversions (`vgic_lr` expects `nat`). This puts a limit on the number of "real" list registers, regardless of whichever `arm_gicvcpu_numlistregs` value you chose. The invariant ensures that these "real" list registers map directly onto `[0, arm_gicvcpu_numlistregs)` (i.e. no truncation).

Reply:
Yeah, so something about this would make a better comment than only repeating in words what the function already does. Roughly speaking we're ensuring that the state relation is a function by clamping the abstract side to the storage size of the concrete side. As you can see this wasn't obvious to either of us, and would make a better comment.

Reply:
@Xaphiosis updated